code/modules/jobs/job_types/bos.dm

```diff
@@ -712,10 +717,15 @@ Scribe
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/buffout)
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/steady)
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/AER9)
+	H.mind.teach_crafting_recipe(/datum/crafting_recipe/AEP7)
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/booster)
 	ADD_TRAIT(H, TRAIT_SURGERY_MID, src)
 	ADD_TRAIT(H, TRAIT_MEDICALGRADUATE, src)
 	ADD_TRAIT(H, TRAIT_CHEMWHIZ, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_ONE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_TWO, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_THREE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_FOUR, src)
```

Comment:
for(var/datum/crafting_recipe/recipe as() in GLOB.knight_gunsmith_recipes)
		H.mind.teach_crafting_recipe(recipe)